datatypes/src/primitives/time_interval.rs

```diff
@@ -167,7 +167,8 @@ impl TimeInterval {
     /// ```
     ///
     pub fn contains(&self, other: &Self) -> bool {
-        self.start <= other.start && self.end >= other.end
+        // TODO: should contains cover equals?
```

Comment:
I guess we should just conform to standardized semantics https://en.wikipedia.org/wiki/Allen%27s_interval_algebra

Reply:
there is no contains in the algebra. So I guess you would use `A == B OR A during B` instead of contains in an algorithm.

Reply:
ok so lets keep it like that. We use it like this at the moment.